clients/src/main/java/org/apache/kafka/common/metrics/Sensor.java

```diff
@@ -224,8 +224,9 @@ public void add(CompoundStat stat) {
      */
     public synchronized void add(CompoundStat stat, MetricConfig config) {
         this.stats.add(Utils.notNull(stat));
+        Object lock = new Object();
         for (NamedMeasurable m : stat.stats()) {
-            KafkaMetric metric = new KafkaMetric(this, m.name(), m.stat(), config == null ? this.config : config, time);
+            KafkaMetric metric = new KafkaMetric(lock, m.name(), m.stat(), config == null ? this.config : config, time);
```

Comment:
Good to mention this change in the PR description since it's an unrelated bug fix.

Reply:
Done.